packages/components/src/snackbar/test/index.tsx

```diff
@@ -0,0 +1,263 @@
+		const testIcon = (
+			<SVG data-testid="icon">
+				<Path />
+			</SVG>
+		);
+
+		render( <Snackbar icon={ testIcon }>Message</Snackbar> );
+
+		const snackbar = screen.getByTestId( testId );
+		const icon = within( snackbar ).getByTestId( 'icon' );
+
+		expect( icon ).toBeInTheDocument();
+	} );
+
+	it( 'should be dismissible by clicking the snackbar', async () => {
+		const onRemove = jest.fn();
+		const onDismiss = jest.fn();
+
+		render(
+			<Snackbar onRemove={ onRemove } onDismiss={ onDismiss }>
+				Message
+			</Snackbar>
+		);
+
+		const snackbar = screen.getByTestId( testId );
+
+		expect( snackbar ).toHaveAttribute( 'role', 'button' );
+		expect( snackbar ).toHaveAttribute(
+			'aria-label',
+			'Dismiss this notice'
+		);
+
+		await click( snackbar );
+
+		expect( onRemove ).toHaveBeenCalled();
+		expect( onDismiss ).toHaveBeenCalled();
+	} );
+
+	it( 'should not be dismissible by clicking the snackbar when the `explicitDismiss` prop is set to `true`', async () => {
+		const onRemove = jest.fn();
+		const onDismiss = jest.fn();
+
+		render(
+			<Snackbar
+				explicitDismiss
+				onRemove={ onRemove }
+				onDismiss={ onDismiss }
+			>
+				Message
+			</Snackbar>
+		);
+
+		const snackbar = screen.getByTestId( testId );
+
+		expect( snackbar ).not.toHaveAttribute( 'role', 'button' );
+		expect( snackbar ).not.toHaveAttribute(
+			'aria-label',
+			'Dismiss this notice'
+		);
+		expect( snackbar ).toHaveClass(
+			'components-snackbar-explicit-dismiss'
+		);
+
+		await click( snackbar );
+
+		expect( onRemove ).not.toHaveBeenCalled();
+		expect( onDismiss ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should be dismissible by clicking the close button when the `explicitDismiss` prop is set to `true`', async () => {
+		const onRemove = jest.fn();
+		const onDismiss = jest.fn();
+
+		render(
+			<Snackbar
+				explicitDismiss
+				onRemove={ onRemove }
+				onDismiss={ onDismiss }
+			>
+				Message
+			</Snackbar>
+		);
+
+		const snackbar = screen.getByTestId( testId );
+		const closeButton = within( snackbar ).getByRole( 'button', {
+			name: 'Dismiss this notice',
+		} );
+
+		await click( closeButton );
+
+		expect( onRemove ).toHaveBeenCalled();
+		expect( onDismiss ).toHaveBeenCalled();
+	} );
+
+	describe( 'actions', () => {
+		it( 'should render only the first action with a warning when multiple actions are passed', () => {
+			render(
+				<Snackbar
+					actions={ [
+						{ label: 'One', url: 'https://example.com' },
+						{ label: 'Two', url: 'https://example.com' },
+						{ label: 'Three', url: 'https://example.com' },
+					] }
+				>
+					Message
+				</Snackbar>
+			);
+
+			expect( console ).toHaveWarnedWith(
+				'Snackbar can only have one action. Use Notice if your message requires many actions.'
+			);
+
+			const snackbar = screen.getByTestId( testId );
+			const actions = within( snackbar ).getAllByRole( 'link' );
+
+			expect( actions ).toHaveLength( 1 );
+			expect( actions[ 0 ] ).toHaveTextContent( 'One' );
```

Comment:
If there's a single element that matches the query, we don't need to use a multiple elements query. A single element query will fail if there are multiple ones anyway, so we're good.

```suggestion
			const action = within( snackbar ).getByRole( 'link' );

			expect( action ).toBeVisible();
			expect( action ).toHaveTextContent( 'One' );
```